btree/btree.c

```diff
@@ -5358,6 +5470,215 @@ static void m0_btree_ut_mt_mt_kv_oper(void)
 	btree_ut_num_threads_num_trees_kv_oper(0, 0);
 }
 
+
+#define MIN_TREE_LOOPS          5000
+#define MAX_TREE_LOOPS          15000
+#define MAX_RECS_FOR_TREE_TEST  100
+
+/**
+ * This routine is a thread handler which primarily involves in creating,
+ * opening, closing and destroying btree. To run out-of-sync with other threads
+ * it also launches PUT, GET, ITER and DEL operations on the btree for a random
+ * count.
+ */
+static void btree_ut_tree_oper_thread_handler(struct btree_ut_thread_info *ti)
+{
+	uint64_t               key;
+	uint64_t               value;
+	m0_bcount_t            ksize = sizeof key;
+	m0_bcount_t            vsize = sizeof value;
+	void                  *k_ptr = &key;
+	void                  *v_ptr = &value;
+	struct m0_btree_rec    rec   = {
+				     .r_key.k_data = M0_BUFVEC_INIT_BUF(&k_ptr,
+									&ksize),
+				     .r_val        = M0_BUFVEC_INIT_BUF(&v_ptr,
+									&vsize),
+				     .r_flags      = 0,
+				     };
+	struct cb_data         data  = {
+					.key         = &rec.r_key,
+					.value       = &rec.r_val,
+					.check_value = false,
+					.flags       = 0,
+				       };
+	struct m0_btree_cb     ut_cb   = {
+					  .c_act       = btree_kv_put_cb,
+					  .c_datum     = &data,
+					 };
+	int32_t                loop_count;
+	struct m0_btree_op     kv_op     = {};
+	void                  *temp_node;
+	struct m0_btree_type   btree_type  = {.tt_id = M0_BT_UT_KV_OPS,
+					      .ksize = sizeof(key),
+					      .vsize = sizeof(value),
+					     };
+	const struct node_type *nt         = &fixed_format;
+	struct m0_be_tx        *tx         = NULL;
+
+	random_r(&ti->ti_random_buf, &loop_count);
+	loop_count %= (MAX_TREE_LOOPS - MIN_TREE_LOOPS);
+	loop_count += MIN_TREE_LOOPS;
+
```

Comment:
It will be good to add transaction apis (like you have added for KV oper handler), so that we do not miss it later.

Reply:
I am planning to add that as a separate task since that will require some more changes. I had just started them with no definite plan.